ptvsd/wrapper.py

```diff
@@ -555,8 +555,11 @@ def close(self):
         self.event_loop_thread.join(WAIT_FOR_THREAD_FINISH_TIMEOUT)
 
         if self.socket:
-            self.socket.shutdown(socket.SHUT_RDWR)
-            self.socket.close()
+            try:
+                self.socket.shutdown(socket.SHUT_RDWR)
+                self.socket.close()
+            except:
+                pass
```

Comment:
Knowing that closing the socket errored-out would be useful, particularly as the ptvsd code is still undergoing so many changes.  Regardless, masking errors is typically a recipe for pain.  So it may be worth logging something here.  At the very least we should emit a warning (or even just print a message).

Reply:
> So it may be worth logging something here. At the very least we should emit a warning (or even just print a message). 

We are printing these, and that's what I'm trying to avoid. We shouldn't be displaying this to the user. Else it looks as though something has gone wrong in the user code or debugger has crashed (that was my initial assumption).

Is it possible that VSC closes the socket before PTVSD gets to this block of code.

> So it may be worth logging something here

Would have, but we don't have any logging mechanism in PTVSD yet.